experimental/dds/attributable-map/src/test/mocha/map.spec.ts

```diff
@@ -28,7 +28,8 @@ function createConnectedMap(
 	id: string,
 	runtimeFactory: MockContainerRuntimeFactory,
 	options?: IMapOptions,
-): AttributableMap {
+	useTestSharedMap: boolean = false,
```

Comment:
not a huge deal, but I would probably just make this test suite always use `TestSharedMap`: it's a trivial extension of the real one that exists only to expose protected functionality.

If you wanted to make it customizable, a strategy that would give you better typechecking would be to take in the class type as a parameter a la what mixins do. As said above, I don't think that's really worth it here though.